airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py

```diff
@@ -197,13 +199,25 @@ def clear_not_launched_queued_tasks(self, session: Session = NEW_SESSION) -> Non
             assert self.kube_client
         from airflow.models.taskinstance import TaskInstance
 
+        default_executor = str(ExecutorLoader.get_default_executor_name())
+
         with Stats.timer("kubernetes_executor.clear_not_launched_queued_tasks.duration"):
             self.log.debug("Clearing tasks that have not been launched")
-            query = select(TaskInstance).where(
-                TaskInstance.state == TaskInstanceState.QUEUED, TaskInstance.queued_by_job_id == self.job_id
-            )
+
+            criteria = [
+                TaskInstance.state == TaskInstanceState.QUEUED,
+                TaskInstance.queued_by_job_id == self.job_id,
+            ]
             if self.kubernetes_queue:
-                query = query.where(TaskInstance.queue == self.kubernetes_queue)
+                criteria.append(TaskInstance.queue == self.kubernetes_queue)
+            elif KUBERNETES_EXECUTOR == default_executor:
+                criteria.append(
+                    or_(TaskInstance.executor == KUBERNETES_EXECUTOR, TaskInstance.executor.is_(None))
+                )
+            else:
+                criteria.append(TaskInstance.executor == KUBERNETES_EXECUTOR)
+            query = select(TaskInstance).where(*criteria)
```

Comment:
I’d just do

```python
query = select(TaskInstance).where(
    TaskInstance.state == TaskInstanceState.QUEUED,
    TaskInstance.queued_by_job_id == self.job_id,
)
if self.kubernetes_queue:
    query = query.where(TaskInstance.queue == self.kubernetes_queue)
if KUBERNETES_EXECUTOR == default_executor:
    query = query.where(
        or_(
            TaskInstance.executor == KUBERNETES_EXECUTOR,
            TaskInstance.executor.is_(None),
        ),
    )
else:
    query = query.where(TaskInstance.executor == KUBERNETES_EXECUTOR)
```

Easier to read IMO.

Reply:
@uranusjr  done.

PS kubernetes executor check added in elif block as if queue name is provided then no need to check for executors